tailwind.config.js

```diff
@@ -0,0 +1,105 @@
+// tailwind.config.js
+
+import tailwindcssAnimate from "tailwindcss-animate";
+import { colorOpacityPlugin } from "./src/lib/plugins/colorOpacityPlugin";
+import colors from "tailwindcss/colors";
+
+/** @type {import("tailwindcss").Config} */
+module.exports = {
+  content: ["./src/**/*.{js,ts,jsx,tsx}", "./src/styles/tailwind.css"],
+  darkMode: ["class", '[data-theme="dark"]'],
+  plugins: [tailwindcssAnimate, colorOpacityPlugin],
+  corePlugins: {
+    preflight: false,
+  },
+  theme: {
+    colors: {
+      inherit: colors.inherit,
+      current: colors.current,
+      transparent: colors.transparent,
+      black: colors.black,
+      white: colors.white,
```

Comment:
what's the use case of black, white, red, yellow? Are there existing semantic tokens that can replace them?

Reply:
I think we don't need full semantic tokens support. It only base color. We don't need to fully trap ourselves in the concept of Obsidian. It's also tailwind project.

Reply:
Using colors like white and black directly will likely to cause dark theme to break, as well as causing unexpected edge cases for people using non default themes. Do you have a good way to avoid these potential breakage?

Reply:
I test some top theme. It's work well in light/dark mode. Dialog overlay will use the black color. I think you can pull up pr #955, you will know about something.